src/sst/core/baseComponent.cc

```diff
@@ -648,6 +709,19 @@ BaseComponent::registerStatisticCore(SST::Params& params, const std::string& sta
     return statistic;
 }
 
+void
+BaseComponent::performStatisticOutput(StatisticBase* stat, bool endOfSimFlag)
+{
+    Simulation::getSimulation()->getStatisticsProcessingEngine()->performStatisticOutput(stat,endOfSimFlag);
+}
+
+void
+BaseComponent::performGlobalStatisticOutput(bool endOfSimFlag)
```

Comment:
Should components be able to call these with endOfSimFlag=true or is that reserved for the final stat dump done by the core? I would think it should be part of the internal core API and the component API shouldn't have it as an option.

Reply:
No, components should not call it with endOfSimFlag=true.  We can either remove this functionality for elements (it's only used in ember), or I can make the BaseComponent version not take an argument and pass false to the underlying stat engine function.

Reply:
The function is needed, Ariel uses it too. But we should remove the endOfSimFlag from this version and pass false to the stat engine.